packet/endpoints.go

```diff
@@ -32,6 +33,47 @@ type ComponentsResponse struct {
 	Components []Component `json:"components"`
 }
 
+// GetWorkflowsFromTink fetches the list of workflows from tink
+func (c *Client) GetWorkflowsFromTink(hwID string) (tw.WorkflowContextList, error) {
+
+	var result tw.WorkflowContextList
+
+	if hwID == "" {
+		return result, errors.New("missing hardware id")
+	}
+
+	labels := prometheus.Labels{"from": "dhcp"}
+	cacherTimer := prometheus.NewTimer(metrics.CacherDuration.With(labels))
+	metrics.CacherRequestsInProgress.With(labels).Inc()
+	metrics.CacherTotal.With(labels).Inc()
+
+	if c.workflowClient == nil {
+		return result, errors.New("unable to fetch workflow, client does not exist")
+	}
+
+	wc := c.workflowClient.(tw.WorkflowSvcClient)
+	resp, err := wc.GetWorkflowContexts(context.Background(), &tw.WorkflowContextRequest{WorkerId: hwID})
+
+	cacherTimer.ObserveDuration()
+	metrics.CacherRequestsInProgress.With(labels).Dec()
+
+	if err != nil {
+		return result, errors.New("fetching the workflow")
+	}
+
+	j, err := json.Marshal(resp)
+	if err != nil {
+		return result, errors.New("marshalling tink hardware")
+	}
+
+	err = json.Unmarshal(j, &result)
```

Comment:
This pattern seems very weird to me, why are using json as an intermediate representation?

Reply:
I was observing error on return type, let me retry this.

Reply:
done modified.